autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -168,14 +169,23 @@ protected void handleScaleDown(List<DecommissionCandidate> decommissionCandidate
                 .map(nodeFqdn -> hostGroupFqdnsToInstanceId.get(nodeFqdn))
                 .collect(Collectors.toList());
 
-        LOGGER.info("ScaleDown NodeCount '{}' for Cluster '{}', HostGroup '{}', NodeIds '{}'",
-                decommissionHostGroupNodeIds.size(), cluster.getStackCrn(), loadAlert.getScalingPolicy().getHostGroup(),
-                decommissionHostGroupNodeIds);
 
         ScalingEvent scalingEvent = new ScalingEvent(loadAlert);
         if (!decommissionHostGroupNodeIds.isEmpty()) {
+            LOGGER.info("ScaleDown NodeCount '{}' for Cluster '{}', HostGroup '{}', NodeIds '{}'",
+                    decommissionHostGroupNodeIds.size(), cluster.getStackCrn(), policyHostGroup,
+                    decommissionHostGroupNodeIds);
+
             scalingEvent.setDecommissionNodeIds(decommissionHostGroupNodeIds);
             eventPublisher.publishEvent(scalingEvent);
+        } else if (existingHostGroupSize > loadAlertConfiguration.getMaxResourceValue()) {
```

Comment:
Similarly here. This check before the actual scale down. Actually the combined check likely needs to happen before processing results from YARN.

Reply:
commented above.